src/verinfast/cloud/aws/instances.py

```diff
@@ -145,8 +146,15 @@ def get_instance_utilization(
     return data
 
 
-def get_instances(sub_id: int, path_to_output: str = "./",
-                  dry=False, log=None) -> str | None:
+def get_instances(sub_id: str, profile: str = None, log=None,
+                  path_to_output: str = "./", dry=False):
+    if profile is None:
+        profile = find_profile(targeted_account=sub_id, log=log)
```

Comment:
This sets the profile from find_profile, but it's not used below.

find_profile uses subprocess to call get-caller-identity and loop through to find the matching one. So does this file using boto3 down below. So the loop in this file is redundant. Same for blocks.py.